### PR TITLE
OP-264 Fixed FlashbagInterface bug, added phone number form validator and updated Readme."

### PR DESCRIPTION
**Pull request by SzymonKostrubiec**
+ Fixed bug with FlashbagInterface
+ Added form validator for phone number with phpSpec tests
+ Update Readme

### CODE REVIEW

File: src/Resources/translations/validators.en.yml [hunk 1]
```diff
@@ -2,3 +2,6 @@ bitbag_sylius_inpost_plugin:
     order:
         phone_number_is_required: 'Phone number is required in order to use InPost shipping method.'
         not_allowed_payment_method: 'Cash on delivery is not allowed to use with InPost shipping method.'
+        phone_number:
```
[the remark this answers is not in the document]
Comment (SzymonKostrubiec) [reply src/Resources/translations/validators.en.yml:5]: No, we don't have polish translations in this plugin

File: src/Validator/HasValidPhoneNumberInPostOrderValidator.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    {
+        $this->shippingMethodChecker = $shippingMethodChecker;
+    }
+
+    /* @param mixed $value */
+    public function validate($value, Constraint $constraint): void
+    {
+        if (!$value instanceof OrderInterface) {
+            throw new \Exception('Value must be instance of OrderInterface');
+        }
```
Comment (senghe) [inline src/Validator/HasValidPhoneNumberInPostOrderValidator.php:34]: `Assert::instanceOf($value, OrderInterface::class);`

File: src/Validator/HasValidPhoneNumberInPostOrderValidator.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+        if (!$value instanceof InPostPointsAwareInterface) {
+            throw new \Exception('Value must be instance of InPostPointsAwareInterface');
```
Comment (senghe) [inline src/Validator/HasValidPhoneNumberInPostOrderValidator.php:37]: Similar as above

**Review by senghe — CHANGES_REQUESTED** [review]

**Review by senghe — APPROVED** [review]

File: src/Validator/HasValidPhoneNumberInPostOrderValidator.php [hunk 4]
```diff
@@ -44,21 +44,32 @@ public function validate($value, Constraint $constraint): void
...
 
     private function addPhoneNumberViolation(string $message): void
     {
         $violationBuilder = $this->context->buildViolation($message);
-        $violationBuilder->setParameter('{{ limit }}', (string) HasValidPhoneNumberInPostOrder::POLISH_PHONE_NUMBER_DEFAULT_LENGTH);
         $violationBuilder->atPath('shippingAddress.phoneNumber')
             ->addViolation();
     }
+
+    private function formatPhoneNumber($phone): string
```
Comment (senghe) [inline src/Validator/HasValidPhoneNumberInPostOrderValidator.php:63]: Missing parameter type

File: src/Validator/Constraint/HasValidPhoneNumberInPostOrder.php [hunk 5]
```diff
@@ -14,12 +14,19 @@
...
 final class HasValidPhoneNumberInPostOrder extends Constraint
 {
-    public const PHONE_NUMBER_IS_TOO_SHORT_MESSAGE = 'bitbag_sylius_inpost_plugin.order.phone_number.min';
-
-    public const PHONE_NUMBER_IS_TOO_LONG_MESSAGE = 'bitbag_sylius_inpost_plugin.order.phone_number.max';
+    public const PHONE_NUMBER_LENGTH_INCORRECT = 'bitbag_sylius_inpost_plugin.order.phone_number.incorrect_length';
 
     public const POLISH_PHONE_NUMBER_DEFAULT_LENGTH = 9;
 
+    public const POSSIBLE_POLISH_PHONE_PREFIXES = [
```
Comment (senghe) [inline src/Validator/Constraint/HasValidPhoneNumberInPostOrder.php:21]: I wouldn't call it with POLISH. It's just POSSIBLE_PHONE_PREFIXES.

File: src/Resources/translations/validators.en.yml [hunk 6]
```diff
@@ -3,5 +3,4 @@ bitbag_sylius_inpost_plugin:
         phone_number_is_required: 'Phone number is required in order to use InPost shipping method.'
         not_allowed_payment_method: 'Cash on delivery is not allowed to use with InPost shipping method.'
         phone_number:
-            min: 'Phone number must be at least {{ limit }} characters long'
-            max: 'Phone  number cannot be longer than {{ limit }} characters long'
+            incorrect_length: "Incorrect phone number for Polish shipping methods"
```
Comment (senghe) [inline src/Resources/translations/validators.en.yml:6]: Phone number must consist of 9 digits.

File: src/Validator/Constraint/HasValidPhoneNumberInPostOrder.php [hunk 7]
```diff
@@ -14,12 +14,19 @@
 
 final class HasValidPhoneNumberInPostOrder extends Constraint
 {
-    public const PHONE_NUMBER_IS_TOO_SHORT_MESSAGE = 'bitbag_sylius_inpost_plugin.order.phone_number.min';
-
-    public const PHONE_NUMBER_IS_TOO_LONG_MESSAGE = 'bitbag_sylius_inpost_plugin.order.phone_number.max';
+    public const PHONE_NUMBER_LENGTH_INCORRECT = 'bitbag_sylius_inpost_plugin.order.phone_number.incorrect_length';
 
     public const POLISH_PHONE_NUMBER_DEFAULT_LENGTH = 9;
```
Comment (senghe) [inline src/Validator/Constraint/HasValidPhoneNumberInPostOrder.php:19]: Please remove the POLISH :pray:

**Review by senghe — CHANGES_REQUESTED** [review]

**Review by senghe — APPROVED** [review]

**Pull request merged**